services/datamanager/builtin/capture_test.go

```diff
@@ -247,27 +247,36 @@ func TestSwitchResource(t *testing.T) {
 	})
 	test.That(t, err, test.ShouldBeNil)
 
-	initialData, err := getSensorData(captureDir)
+	dataBeforeSwitch, err := getSensorData(captureDir)
 	test.That(t, err, test.ShouldBeNil)
 
 	passTimeCtx2, cancelPassTime2 := context.WithCancel(context.Background())
 	donePassingTime2 := passTime(passTimeCtx2, mockClock, captureInterval)
 
 	// Test that sensor data is captured from the new collector.
 	waitForCaptureFilesToExceedNFiles(captureDir, len(getAllFileInfos(captureDir)))
-	testFilesContainSensorData(t, captureDir, func(t *testing.T, sd []*v1.SensorData) {
-		valueMap := map[float64]int{}
-		for _, d := range sd {
-			jmag := d.GetStruct().GetFields()["Number"].GetStructValue().GetFields()["Dual"].GetStructValue().GetFields()["Jmag"].GetNumberValue()
-			valueMap[jmag]++
-		}
-		// Each resource's mocked capture method outputs a different value. Assert that we see data captured by the initial arm1 resource as well as the changed resource.
-		test.That(t, len(valueMap), test.ShouldEqual, 2)
-		test.That(t, valueMap[float64(444)], test.ShouldBeGreaterThan, 0)
+	testFilesContainSensorData(t, captureDir)
 
-		// Assert that the initial arm1 resource isn't capturing any more data.
-		test.That(t, valueMap[float64(1)], test.ShouldEqual, len(initialData))
-	})
+	filePaths := getAllFilePaths(captureDir)
+	test.That(t, len(filePaths), test.ShouldEqual, 2)
+
+	initialData, err := datacapture.SensorDataFromFilePath(filePaths[0])
+	test.That(t, err, test.ShouldBeNil)
+	// Assert that the initial arm1 resource isn't capturing any more data.
```

Comment:
[supernit] Just for organization here and below, would add this check after the for loop, since that verifies that file1 only contains data from the initial arm1 resource. Then, we can double check that it didn't collect any more data after the reconfiguration resource switch.

Reply:
Done!